homework/questions.txt

```diff
@@ -3,72 +3,134 @@ we went through in class.
 
 1. Navigating the terminal
   a, What is a root directory?
+  The root directory is the first or top-most directory in a hierarchy. It can be likened to the trunk of a tree, as the starting point where all branches originate from.
+  
 
   b, What is the difference between root and home directory?
+  The root directory is the topmost level of the system drive and it is denoted by a slash '/'. The home directory is a subdirectory of the root directory and it is denoted by '~' and has path "/users/username".
+  
 
   c, What is the difference between absolute and relative path?
+  An absolute path is defined as the specifying the location of a file or directory from the root directory(/) and Relative path is defined as the path related to the present working directly(pwd). It starts at your current directory and never starts with a /.
+  
 
   d, What command do you use to see what directory are you currently in?
+  pwd (print working directory
+  
 
   e, How would you see the files and folders in a directory?
+  ls
+  
 
   f, How do you go back to your home directory?
+  cd~
+  
 
   g, How do you create a new folder, can you also specify in the command where you want to save it ?
+  cd my documents
+  To create a folder in your present directory you just need to use the command mkdir "name of the folder", but if you want to create a folder in other directory you just need to use mkdir path/path/name of the folder.
 
   h, How do you remove a folder/directory, is it the same as removing a file?
+  No, to remove an empty folder we need to use the command-line rmdir "name of the folder".
+  
 
   i, What is a file or folder is not empty ? What flag would you use to remove it anyway?
+  rm -rf "name of the folder"
+  
 
   j, How do you move a folder/file ? What arguments does the command take?
+  mv "name of the file" "name of the folder"/"name of the file"
+  
 
   k, How can you find out more about a command and see the available options, syntax etc.
     (How/Where do you see the manual/help for a command)
+    man "command"
+    
 
   l, With what command do you rename a folder/file in the command line?
+  mv "name of the file/folder" "new name of the file/folder".
+  
 
   m, How do you copy a folder ?
+  cp -r "name of the folder" "name of the folder copy"
 
   n, What command do you use to open a folder/file ?
+  cat or less
 
 2. Creating a new project
   a, How do you create a new folder (what command) ?
+  mkdir "name of the folder"
+  
 
   b, What command do you use to create/initiate a local repository (from the command line)?
      /One can also create a new repository on Github and then just clone it to their machine./
+     git init
+     
 
   c, What command do you use to create a new file ?
+  touch "name of the file"
+  
 
   d, If you create a file and and add text to it, these changes are now on your working directory.
      How would you save these to the local repository (describe the steps with commands)
+     fork
```

Comment:
if you are creating a project on your local directory, you dont need to fork anything. You could create an empty repository on your github and then clone it. But nothing to fork, as that is a command for copy on github.